### ISSUE TITLE
Multiple bypasses by Shivam Bathla Vol. 2

### ISSUE THREAD

**Issue by franbuehler** [coreruleset/coreruleset#2924]
### Description

On our CRS share in the folder "Security 2022" we have another list of bypasses submitted by Shivam Bathla that we should address.
Like those addressed in #2626.

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by franbuehler** [comment]
PRs addressing this bypasses are:

* https://github.com/coreruleset/coreruleset/pull/2925
* https://github.com/coreruleset/coreruleset/pull/2926

**Comment by franbuehler** [comment]
New PR: https://github.com/coreruleset/coreruleset/pull/3019

**Comment by franbuehler** [comment]
According to Shivam's second bypass list, at least the following rules require a stricter sibling. If these are covered, 33 out of 42 bypasses are detected (9 not).

930 LFI
- [ ] 930120

932 RCE
- [ ] 932150
- [ ] 932160

942 SQLi
- [ ] 942151
- [ ] 942320

PL2 rules that could be extended with targets: User-Agent and Referer:


932 RCE
- [ ] 932125
- [ ] 932200
- [ ] 932230
- [ ] 932235
- [ ] 932240


942 SQLi

- [ ] 942110
- [ ] 942200
- [ ] 942370
- [ ] 942410
- [ ] 942480
- [ ] 942521

**Comment by dune73** [comment]
Not sure I get you correctly.

The idea is to add PL2 stricter sibling including UA + Referer for some of the rules and existing PL2 rules would get UA + Referer? If we do this, we cover 37 out of 42 of the bypasses. The remaining ones will wait for update in 932150 and then some more work.

Correct?


Also: 932125 is a PL1 rule. (Might have to check the rest as well)

**Comment by franbuehler** [comment]
Yes, that's correct. I'm not sure if the changes of 932150 will cover 2 more of the currently 5 (42-37) missing bypasses.
I'll have to check after the merge.

That's possible. I'll see this when touching the rules. Thank you for the hint.

**Comment by dune73** [comment]
Got you. Thank you for the explanation. I think this is the right course here.

**Comment by theseion** [comment]
I've ticked the box for 932150. That's all done now.

932160 should be good to go, if you want to add a sibling for that one too.

Thanks for the work you did here!

**Comment by franbuehler** [comment]
Thank you @theseion!!

I can confirm that from 42 reported bypasses (request headers), 39 bypasses are now detected at PL2.
He twice tests with the request header `Accept-Encoding`, but we defined that we will add request headers `Referer` and `User-Agent`.
3 remaining PoCs are not detected, they're just too short I think. I can share them in our private Slack channel.

If we decide:

1. we leave those 3 PoCs (shared in Slack) undetected and
2. ignore the `Accept-Encoding` header

the work is completed and this issue can be closed.

**Comment by dune73** [comment]
I think we can give it a shot at PL2 or PL3 if you are skeptical.

Last few years saw a lot of UA / Referer attacks, we have been quite laid back with regards to that. I think we have to expect more RCEs on these header fields in the future and the probes in question should be blocked on servers with some serious security needs (and the ability to deal with rare FPs).

**Comment by franbuehler** [comment]
Ok, I added a new task and I'll open a PR this afternoon.

**Comment by theseion** [comment]
This issue looks done. Can we close it?

**Comment by franbuehler** [comment]
Yes, I'll close it.
